lib/bolt/puppetfile/installer.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'r10k/cli'
+require 'bolt/r10k_log_proxy'
+require 'bolt/error'
+
+# This class is used to install modules from a Puppetfile to a module directory.
+#
+module Bolt
+  class Puppetfile
+    class Installer
+      def initialize(path, moduledir, config = {})
+        @path      = path
+        @moduledir = moduledir
+        @config    = config
+        validate
+      end
+
+      def install
```

Comment:
The more I think about it, the more I think it makes sense for `path` and `moduledir` to be arguments to `install` instead of instance variables. If you consider the lifecycle of this object, it's possible it would be constructed early on during configuration, before we know where/what we want to install with it.